src/java.base/share/classes/java/util/DualPivotQuicksort.java

```diff
@@ -362,73 +375,182 @@ && tryMergeRuns(sorter, a, low, size)) {
-                 * location formerly occupied by the pivot. After
-                 * completion of partitioning the pivot is swapped
-                 * back into its final position, and excluded from
-                 * the next subsequent sorting.
+                 * Sort the right part (possibly in parallel), excluding
+                 * known pivot. All elements from the central part are
+                 * equal and therefore already sorted.
                  */
-                a[e3] = a[lower];
+                if (size > MIN_PARALLEL_SORT_SIZE && sorter != null) {
+                    sorter.forkSorter(bits | 1, upper, high);
+                } else {
+                    sort(sorter, a, bits | 1, upper, high);
+                }
+            }
+            high = lower; // Iterate along the left part
+        }
+    }
 
-                /*
-                 * Traditional 3-way (Dutch National Flag) partitioning
-                 *
-                 *   left part                 central part    right part
-                 * +------------------------------------------------------+
-                 * |   < pivot   |     ?     |   == pivot   |   > pivot   |
-                 * +------------------------------------------------------+
-                 *              ^           ^                ^
-                 *              |           |                |
-                 *            lower         k              upper
-                 *
-                 * Invariants:
-                 *
-                 *   all in (low, lower] < pivot
-                 *   all in (k, upper)  == pivot
-                 *   all in [upper, end] > pivot
-                 *
-                 * Pointer k is the last index of ?-part
-                 */
-                for (int k = ++upper; --k > lower; ) {
-                    int ak = a[k];
+    /**
+     * Partitions the specified range of the array using the two pivots provided.
+     *
+     * @param array the array to be partitioned
+     * @param low the index of the first element, inclusive, for partitioning
+     * @param high the index of the last element, exclusive, for partitioning
+     * @param pivotIndex1 the index of pivot1, the first pivot
+     * @param pivotIndex2 the index of pivot2, the second pivot
+     *
+     */
+    @ForceInline
+    private static int[] partitionDualPivot(int[] a, int low, int high, int pivotIndex1, int pivotIndex2) {
+        int end = high - 1;
+        int lower = low;
+        int upper = end;
 
-                    if (ak != pivot) {
-                        a[k] = pivot;
+        int e1 = pivotIndex1;
+        int e5 = pivotIndex2;
+        int pivot1 = a[e1];
+        int pivot2 = a[e5];
 
-                        if (ak < pivot) { // Move a[k] to the left side
-                            while (a[++lower] < pivot);
+        /*
+        * The first and the last elements to be sorted are moved
```

Comment:
Misaligned `/*` here and several places later. `*` should be aligned.

Reply:
Fixed the alignment for `/*` and `*` in various places. Please see the latest commit that was pushed.